native/c/zds.hpp

```diff
@@ -55,18 +55,62 @@ typedef struct
   std::string vol;       // Volume Serial
 } DS_ATTRIBUTES;
 
-std::string zds_get_recfm(const fldata_t& file_info);
+std::string zds_get_recfm(const fldata_t &file_info);
 
-/**
- * @brief Read data from a z/OS data set
- *
- * @param zds data set returned attributes and error information
- * @param dsn data set name from which to read
- * @param response data read
- * @param encoding The desired encoding for the data set (optional)
- * @return int 0 for success; non zero otherwise
- */
-int zds_read_from_dsn(ZDS *zds, std::string dsn, std::string &response);
+extern "C"
```

Comment:
We don't always want `extern "C"` so we should wrap it in a conditional macro for creating Python bindings.

I know I left a similar comment on the related PR, not sure which one we want to handle it in 😋

Reply:
I'm trying to help by doing some testing with LINKEDIT options on the other PR.
Once I have a working solution, I don't mind helping with this here as well.